SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -2261,6 +2271,11 @@ void USpatialReceiver::ResolveIncomingOperations(UObject* Object, const FUnrealO
 	UE_LOG(LogSpatialReceiver, Verbose, TEXT("Resolving incoming operations depending on object ref %s, resolved object: %s"),
 		   *ObjectRef.ToString(), *Object->GetName());
 
+	/* Rep-notify can modify ObjectRefToRepStateMap in some situations which can cause the TSet to access
```

Comment:
This is the assumption. Has it been observed? If it has not, can we add some ensure in relevant places to confirm that we are seeing a re-entrancy issue ?

Reply:
So whilst it's not been observed there is at least one clear chain of events which would cause strangeness (worker->worker replication):

USpatialReceiver::CleanupRepStateMap
USpatialNetDriver::RemoveActorChannel
USpatialActorChannel::Close
UNetDriver::NotifyActorDestroy
UWorld::DestroyActor
[RepNotify]
USpatialReceiver::ResolveIncomingOperations
USpatialReceiver::ResolvePendingOperations
USpatialReceiver::ReceiveActor
USpatialReceiver::LeaveCriticalSection
USpatialReceiver::OnCriticalSection
USpatialDispatcher::ProcessOps
USpatialNetDriver::TickDispatch

It's a rare race and potentially not the fix for the stack we saw but it's hard to know!